src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -251,7 +289,10 @@ private[spark] case class CosmosDBConnection(config: Config) extends CosmosDBLog
 
           if (pageCount >= maxPagesPerBatch) {
             if (maxPagesPerBatch > 1) {
-              nextContinuation = previousBlockStartContinuation + "|" + feedItems.last.get("id")
+              // set the continuation to lsn of last processed record - 1 - to make sure
+              // the lsn containing teh last processed record is returned again
+              // so that we can find the last processed record and continue processing from there
+              nextContinuation = (feedItems.last.getInt("_lsn") - 1).toString() + "|" + feedItems.last.get("id")
```

Comment:
is it guaranteed for documents to have lsn? if not this may through NPE.

Reply:
yes - documents always have lsn - it reflects the transaction with which the document was inserted/updated.